src/plots/xyPlot.ts

```diff
@@ -135,15 +237,28 @@ export class XYPlot<X, Y> extends Plot {
     var adjustCallback = key === XYPlot._X_KEY ? this._adjustYDomainOnChangeFromXCallback
                                                : this._adjustXDomainOnChangeFromYCallback;
     scale.onUpdate(adjustCallback);
+
+    if (this._lazyDomainChange) {
+      scale.offUpdate(this._renderCallback);
```

Comment:
Instead of doing this on/off switch, wouldn't it make more sense if `_renderCallback`'s internal behavior switched on `_lazyDomainChange`?